apps/server/src/controllers/ontimeController.validate.ts

```diff
@@ -166,3 +171,137 @@ export const validateLoadProjectFile = [
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ errors: errors.array() });
+    }
+
+    next();
+  },
+];
+
+/**
+ * @description Validates the filenames for renaming a project.
+ */
+export const validateProjectRename = [
+  body('projectFilename')
+    .exists()
+    .withMessage('Project filename is required')
+    .isString()
+    .withMessage('Project filename must be a string'),
+
+  body('newProjectFilename')
+    .exists()
+    .withMessage('Duplicate project filename is required')
+    .isString()
+    .withMessage('Duplicate project filename must be a string'),
+
+  (req, res, next) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ errors: errors.array() });
+    }
+
+    next();
+  },
+];
+
+/**
+ * @description Validates the filename for creating a project file.
+ */
+export const validateProjectCreate = [
+  body('projectFilename')
+    .exists()
+    .withMessage('Filename is required')
+    .isString()
+    .withMessage('Filename must be a string'),
+  (req, res, next) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ errors: errors.array() });
+    }
+
+    next();
+  },
+];
+
+const checkExistingFile = async (projectFilename?: string): Promise<string | null> => {
+  const projectFilePath = join(uploadsFolderPath, projectFilename);
+
+  try {
+    const fh = await open(projectFilePath);
+    close(fh.fd);
+  } catch (error) {
+    return 'Project file does not exist';
+  }
+};
+
+const checkNewFile = async (newProjectFilename: string): Promise<string | null> => {
+  const newProjectFilePath = join(uploadsFolderPath, newProjectFilename);
+
+  try {
+    const fh = await open(newProjectFilePath);
+    close(fh.fd);
+
+    // File exists, so we can't continue
+    return 'New file name already exists';
+  } catch (error) {
+    // File does not exist, so we can continue
+    return null;
+  }
+};
+
+/**
+ * @description Validates the existence of project files.
+ * @param {object} projectFiles
+ * @param {string} projectFiles.projectFilename
+ * @param {string} projectFiles.newProjectFilename
+ *
+ * @returns {Promise<Array<string>>} Array of errors
+ *
+ */
+export const validateProjectFiles = async (projectFiles: {
+  projectFilename?: string;
+  newProjectFilename?: string;
+}): Promise<Array<string>> => {
+  const errors = [];
+  try {
+    if (projectFiles.projectFilename) {
+      const existingFileError = await checkExistingFile(projectFiles.projectFilename);
+
+      if (existingFileError) {
+        errors.push(existingFileError);
+      }
+    }
+
+    if (projectFiles.newProjectFilename) {
+      const newFileError = await checkNewFile(projectFiles.newProjectFilename);
+
+      if (newFileError) {
+        errors.push(newFileError);
+      }
+    }
+
+    return errors;
+  } catch (error) {
+    return [error.toString()];
+  }
```

Comment:
Question: I am not sure I understand the reason to re-throw if we dont enrich the error. Maybe I am missing something?

Reply:
yeah you're right this is silly